### PR TITLE
T2115

### PR DESCRIPTION
**Pull request by secretrobotron**

### CODE REVIEW

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @humphd received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/97c204e33a54ba6/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @humphd received. Current queue size: 1

Live output at: http://botio.butyr.org:8000/9bac13f2f8885e6/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/97c204e33a54ba6/output.txt

Total script time: 1.75 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/9bac13f2f8885e6/output.txt

Total script time: 1.52 mins

Server is running on http://pr1039.butyr.org/templates/basic/
Run the Test Suite: http://pr1039.butyr.org/test/

File: src/timeline/track-container.js [hunk 1]
```diff
@@ -41,36 +43,52 @@ define( [ "core/logger", "util/dragndrop", "./trackevent-drag-manager" ],
...
-        // ensure its a plugin and that only the area under the last track is droppable
+        // Ensure its a plugin and that only the area under the last track is droppable
         if ( mousePosition[ 1 ] > lastTrackBottom ) {
-          var newTrack = butter.currentMedia.addTrack(),
-              trackRect = newTrack.view.element.getBoundingClientRect(),
-              left = mousePosition[ 0 ] - trackRect.left,
-              start = left / trackRect.width * newTrack.view.duration,
-              draggableType = ( dropped.element ? dropped.element : dropped ).getAttribute( "data-butter-draggable-type" );
+
+          draggableType = ( dropped.element ? dropped.element : dropped ).getAttribute( "data-butter-draggable-type" );
```
Comment (humphd) [inline src/timeline/track-container.js]:
``` javascript
draggableType = ( dropped.element || dropped ).getAttribute...
```

File: src/timeline/track-container.js [hunk 2]
```diff
@@ -41,36 +43,52 @@ define( [ "core/logger", "util/dragndrop", "./trackevent-drag-manager" ],
...
               track: newTrack,
               type: dropped.getAttribute( "data-popcorn-plugin-type" )
             });
-          } else if ( draggableType === "trackevent" ) {
-            newTrack.view.dispatch( "trackeventdropped", {
-              start: dropped.data.start,
-              track: newTrack,
-              trackEvent: dropped.data.trackEvent
+          }
+          else if ( draggableType === "trackevent" ) {
```
Comment (dseif) [inline src/timeline/track-container.js:76]: `} else if` pretty please

Comment (secretrobotron) [inline src/timeline/track-container.js:76]: I actually use the vertically longer form whenever I code. You really don't like it? I find it much easier to digest.

**Pull request merged**